commit c5cde9050dfbec72617f6417cc2e5c38094ca0d0
Author: Simon Lelievre <simon.lelievre@corp.ovh.com>
Date:   Mon Apr 4 17:30:13 2016 +0200

    return status code in liolib pclose

diff --git a/src/liolib.c b/src/liolib.c
index 649f9a5..e7a7105 100644
--- a/src/liolib.c
+++ b/src/liolib.c
@@ -109,7 +109,20 @@ static int io_pclose (lua_State *L) {
   FILE **p = tofilep(L);
   int ok = lua_pclose(L, *p);
   *p = NULL;
-  return pushresult(L, ok, NULL);
+
+  int en = errno;  /* calls to Lua API may change this value */
+
+  if ( ok != -1 ) {
+    lua_pushboolean(L, 1);
+    lua_pushstring(L, "");
+    lua_pushinteger(L, 0);
+  } else {
+    lua_pushnil(L);
+    lua_pushfstring(L, "%s", strerror(en));
+    lua_pushinteger(L, en);
+  }
+  lua_pushinteger(L, ok >> 8);
+  return 4;
 }
 
 
diff --git a/src/luaconf.h b/src/luaconf.h
index 7102028..84d3c4d 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -629,7 +629,7 @@ union luai_Cast { double l_d; long l_l; };
 #if defined(LUA_USE_POPEN)
 
 #define lua_popen(L,c,m)	((void)L, fflush(NULL), popen(c,m))
-#define lua_pclose(L,file)	((void)L, (pclose(file) != -1))
+#define lua_pclose(L,file)	((void)L, pclose(file))
 
 #elif defined(LUA_WIN)
 
